spdmlib/src/crypto/x509v3.rs

```diff
@@ -1223,7 +1283,7 @@ mod tests {
         assert!(check_leaf_certificate(&end1, true).is_ok());
         assert!(check_leaf_certificate(&end1, false).is_ok());
         assert!(check_leaf_certificate(&end2, false).is_ok());
-        assert!(check_leaf_certificate(&end2, true).is_err());
+        assert!(check_leaf_certificate(&end2, true).is_ok());
```

Comment:
end2 contains hardware oid and alias cert shall not have hardware oid. Need return Err.